classes/local/store/s3/client.php

```diff
@@ -79,44 +79,91 @@ public function __wakeup() {
         // it will be serialised, so re-retrive them now.
         $config = manager::get_objectfs_config();
         $this->set_client($config);
-        $this->client->registerStreamWrapper();
+        if ($this->is_functional()) {
+            $this->client->registerStreamWrapper();
+        }
     }
 
-    public function set_client($config) {
-        $options = array(
-            'region' => $config->s3_region,
-            'version' => AWS_API_VERSION
-        );
+    /**
+     * Check if the client is functional.
+     * @return bool
+     */
+    private function is_functional() {
+        return !is_null($this->client);
```

Comment:
Saves an op. That and isset being a language construct it should be slightly faster than a function call.

```suggestion
        return isset($this->client);
```

Reply:
yeah agree. isset would be better here. changing...